addon/model-data.js

```diff
@@ -121,11 +121,20 @@ export default class M3ModelData {
 
   didCommit(data) {
     let changedKeys;
-    if (data) {
+    if (data && data.attributes) {
       changedKeys = this._mergeUpdates(data.attributes, commitDataAndNotify);
       this._notifyProjectionProperties(changedKeys);
     }
 
+    let hasReceivedId = !this.id && data && data.id;
+    if (hasReceivedId) {
```

Comment:
Seems like this can be `if (hasReceivedId && this.baseModelName)`

Reply:
Not quite - we always want to get the new ID (`this.id = coerceId(data.id)`) and if there is base model name, do the rest.